src/main/org/firebirdsql/gds/ng/StatementOperationAware.java

```diff
@@ -0,0 +1,51 @@
+package org.firebirdsql.gds.ng;
+
+import org.firebirdsql.gds.Operation;
+
+/**
+ * Abstract class for controlling the execution of statement
+ *
+ * @author <a href="mailto:vasiliy.yashkov@red-soft.ru">Vasiliy Yashkov</a>
+ */
+public abstract class StatementOperationAware {
```

Comment:
See also my comment in the JDBC: I think this should be separated in the interface `StatementOperationAware` and a separate class for registering it (and usage by the driver).